components/brave_new_tab_ui/components/search/SearchBox.tsx

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+import Flex from '$web-common/Flex';
+import Dropdown from '@brave/leo/react/dropdown';
+import Icon from '@brave/leo/react/icon';
+import Input from '@brave/leo/react/input';
+import { icon, spacing } from '@brave/leo/tokens/css';
+import { stringToMojoString16 } from 'chrome://resources/js/mojo_type_util.js';
+import { AutocompleteResult, OmniboxPopupSelection, PageHandler, PageHandlerRemote, PageInterface, PageReceiver } from 'gen/components/omnibox/browser/omnibox.mojom.m';
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface SearchEngine {
+  name: string,
+  icon: string
+}
+const searchEngines: SearchEngine[] = [{
+  name: 'Brave',
+  icon: 'brave-icon-search-color'
+}, {
+  name: 'Google',
+  icon: 'google-color'
+}, {
+  name: 'DuckDuckGo',
+  icon: 'duckduckgo-color'
+}, {
+  name: 'Bing',
+  icon: 'bing-color'
+}]
+
+interface Props {
+}
+
+const SearchInput = styled(Input)`
+  --leo-control-padding: 6px;
+
+  display: inline-block;
+  width: 540px;
+`
+
+const EnginePicker = styled(Dropdown)`
+`
+
+const SelectedIcon = styled(Icon)`
+  --leo-icon-size: ${icon.l};
+`
+
+const SearchIconContainer = styled.div`
+  padding-right: ${spacing.m};
+`
+
+const Option = styled.div`
+    display: flex;
+    gap: ${spacing.m};
+`
+
+export const omniboxController: PageHandlerRemote = PageHandler.getRemote()
+  ; (window as any).omnibox = omniboxController;
```

Comment:
Can we remove the `;` at the last?

```suggestion
;(window as any).omnibox = omniboxController
```

Reply:
Heh - that's a weird one - I think VSCode and Prettier were fighting it out. Prettier doesn't like semicolons, so it wants

```
const foo = whatever()
;(6 + 5)
```

because otherwise it'd be equivalent to

```
const foo = whatever()(6 + 5)
```

so it puts the semicolon at the start, which is ugly as. I've put it on the end of the line above because I'm not a fan of the leading semicolon :laughing:

Reply:
Yeah, I agree that `;` between two lines are needed. But not sure why prettier put the `;` after `= omniboxController` too 😄